src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2303,6 +2303,53 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         tmp = nullptr;
     }
 
+    void make_type_conversions(Allocator &al, std::string call_name, Vec<ASR::call_arg_t> args,
+                                const Location &loc) {
```

Comment:
Make this function return an `expr_t*`, and just return the value instead of assigning it to `tmp`. Then assign it to `tmp` in the caller.